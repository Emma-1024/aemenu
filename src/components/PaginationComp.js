import React, { Component } from 'react'
import { Grid, Row, Pagination } from 'react-bootstrap'


class PaginationComp extends Component {
  constructor(props) {
    super(props)
    this.state = {
      active: 1,
      indexObj: { param: 'all', start: 0, end: 0 }
    }
  }

  render() {
    this._forItems()
    return (
      <div>
        <Grid container="true">
          <Row>
            <Pagination bsSize="large">{this.items}</Pagination>
            <br />
          </Row>
        </Grid>
      </div>
    )
  }

  _forItems() {
    this.items = []
    let obj = this.props.data
    let totalData = obj.seasonAllData.length
    let param = obj.seasonParam
    let onePage = obj.end
    let pageNumber = Math.ceil(totalData / onePage)
    for (let number = 1; number <= pageNumber; number++) {
      this.items.push(
        <Pagination.Item
          onClick={() => this._pageClick(param, number, onePage)}
          active={number === this.state.active}
          key={number}
        >
          {number}
        </Pagination.Item>
      )
    }
  }

  _pageClick(param, pageNumber, onePage) {
    // console.log(param)
    let start = (pageNumber - 1) * onePage
    let end = start + onePage
    let obj = { param: param, start: start, end: end }
    this.setState({
      indexObj: obj,
      active: pageNumber
    }, () => {
      this.props.getPageIndex(this.state.indexObj)
    })
  }
}

export default PaginationComp
