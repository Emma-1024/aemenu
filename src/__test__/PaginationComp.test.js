import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import PaginationComp from '../components/PaginationComp'

const compName='PaginationComp'

// test(`${compName} Test`, () => {
//   const div = document.createElement('div')
//   ReactDOM.render(<PaginationComp />, div)
//   ReactDOM.unmountComponentAtNode(div)
// })

const getPageIndex = jest.fn()

test(`${compName} Test`, ()=>{
  const testRenderer = renderer.create(
    <PaginationComp getPageIndex={getPageIndex} data={testData}></PaginationComp>
  )
  const testInstance = testRenderer.root

  let tree = testRenderer.toJSON()
  expect(tree).toMatchSnapshot()

})
