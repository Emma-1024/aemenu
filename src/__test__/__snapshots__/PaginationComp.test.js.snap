// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationComp Test 1`] = `
<div>
  <div
    className="container"
    container="true"
  >
    <div
      className="row"
    >
      <ul
        className="pagination pagination-lg"
      />
      <br />
    </div>
  </div>
</div>
`;
